### PR TITLE
Failure domains support

### PR DESCRIPTION
**Pull request by freeekanayaka**

### CODE REVIEW

File: lxd/db/cluster/schema.go [hunk 1]
```diff
@@ -20,6 +20,11 @@ CREATE TABLE config (
     value TEXT,
     UNIQUE (key)
 );
+CREATE TABLE failure_domains (
```
Comment (stgraber) [inline lxd/db/cluster/schema.go:23]: Given this is tied to nodes, shouldn't it be `nodes_failure_domains`?

File: client/lxd_cluster.go [hunk 2]
```diff
@@ -113,6 +113,11 @@ func (r *ProtocolLXD) UpdateClusterMember(name string, member api.ClusterMemberP
 	if !r.HasExtension("clustering_edit_roles") {
 		return fmt.Errorf("The server is missing the required \"clustering_edit_roles\" API extension")
 	}
+	if member.FailureDomain != "" {
+		if !r.HasExtension("clustering_failure_domains") {
+			return fmt.Errorf("The server is missing the required \"clustering_failure_domains\" API extension")
+		}
+	}
```
Comment (stgraber) [inline client/lxd_cluster.go:120]: Please make that its own commit. Changes to `client/` are always backported.

**Comment by stgraber** [comment]
Shouldn't we be showing `default` as the failure domain for nodes that haven't been configured otherwise?

That would be consistent with what we do for projects and profiles.

File: lxd/db/cluster/schema.go (hunk 1 above)
Comment (freeekanayaka) [reply lxd/db/cluster/schema.go]: Done.

File: client/lxd_cluster.go (hunk 2 above)
Comment (freeekanayaka) [reply client/lxd_cluster.go:120]: Done.

**Comment by stgraber** [comment]
@freeekanayaka did you miss the comment above about "default"?

**Comment by freeekanayaka** [comment]
> @freeekanayaka did you miss the comment above about "default"?

Ah yeah I think so.. I'll nail it tomorrow morning first thing.

**Comment by freeekanayaka** [comment]
> Shouldn't we be showing `default` as the failure domain for nodes that haven't been configured otherwise?
> 
> That would be consistent with what we do for projects and profiles.

Fixed.
